sdk/sdk-api/src/main/java/mil/emp3/api/WMS.java

```diff
@@ -162,13 +160,15 @@ public void setLayers(List<String> newLayers) throws EMP_Exception {
      */
     @Override
     public String toString() {
-        String str = "URL: " + serviceURL + "\n"
+        String str = "URL: " + getURL() + "\n"
                 + "WMS Version: " + eVersion.toString() + "\n"
                 + "Transparent: " + bTransparent + "\n"
-                + "Tile format: " + sTileFormat + "\n"
-                + "Layers: \n";
-        for (String layer : oLayers) {
-            str += "\t" + layer + "\n";
+                + "Tile format: " + sTileFormat + "\n";
+        if (oLayers != null && oLayers.size() > 0) {
+            str += "Layers: \n";
+            for (String layer : oLayers) {
+                str += "\t" + layer + "\n";
+            }
         }
```

Comment:
In the constructor for WMS, eVersion is defaulted to VERSION_1_3_0 if it is null.